speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -285,18 +215,121 @@ public void onError(Throwable t) {}
 
             tempByteString = ByteString.copyFrom(sharedQueue.take());
 
+            checkStopRecognitionFlag(tempByteString.toByteArray());
+
             request =
                 StreamingRecognizeRequest.newBuilder().setAudioContent(tempByteString).build();
 
             audioInput.add(tempByteString);
           }
-
+          Thread.sleep(100);
           clientStream.send(request);
         }
+        clientStream.closeSend();
       } catch (Exception e) {
         System.out.println(e);
       }
     }
   }
+
+  public static ResponseObserver<StreamingRecognizeResponse> getResponseObserver() {
+    return new ResponseObserver<StreamingRecognizeResponse>() {
+
+      final ArrayList<StreamingRecognizeResponse> responses = new ArrayList<>();
+
+      public void onStart(StreamController controller) {
+        referenceToStreamController = controller;
+      }
+
+      public void onResponse(StreamingRecognizeResponse response) {
+        responses.add(response);
+        StreamingRecognitionResult result = response.getResultsList().get(0);
+        Duration resultEndTime = result.getResultEndTime();
+        resultEndTimeInMS =
+                (int) ((resultEndTime.getSeconds() * 1000) + (resultEndTime.getNanos() / 1000000));
+        double correctedTime =
+                resultEndTimeInMS - bridgingOffset + (STREAMING_LIMIT * restartCounter);
+
+        SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
+        if (result.getIsFinal()) {
+          System.out.print("\r");
+          System.out.printf(
+                  "%s: %s [confidence: %.2f]\n",
+                  convertMillisToDate(correctedTime),
+                  alternative.getTranscript(),
+                  alternative.getConfidence());
+          isFinalEndTime = resultEndTimeInMS;
+          lastTranscriptWasFinal = true;
+        } else {
+          System.out.print("\r");
+          System.out.printf(
+                  "%s: %s", convertMillisToDate(correctedTime), alternative.getTranscript());
+          lastTranscriptWasFinal = false;
+        }
+        checkStopRecognitionFlag(alternative.getTranscript().getBytes(StandardCharsets.UTF_8));
```

Comment:
this method will fail if the "exit" word is not in the first element. can such thing happen? can the code reflect that?

Reply:
this code reflects only if the word "exit" will be present in a sentence , that's why we check element length before comparing

Reply:
My question was related to line 256. Is it possible that other, less confident, alternatives interpret "exit" while the first one does not?

Reply:
I think we have some possibility, so in this case, we should repeat "exit".